### PR TITLE
[runtime-security] Execute runtime security functional tests using Kitchen

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Execute the runtime security agent functional tests using Kitchen

### Motivation

Test the runtime security agent on multiple OS / kernel versions

### CODE REVIEW

File: .gitlab-ci.yml
Comment (KSerrania) [inline .gitlab-ci.yml:1770]: I think there's a missing dash here, I'm getting an error when doing:
```
$ inv system-probe.build -no-with-bcc
No idea what '-t' is!
```

```suggestion
    - inv -e system-probe.build --bundle-ebpf --incremental-build --no-with-bcc
```

**Review by KSerrania — COMMENTED** [review]
You may want to make the agent-security team owner of `/test/kitchen/site-cookbooks/dd-agent-runtime-security-check/` and `/test/kitchen/kitchen-azure-runtime-security-test.yml` in the `CODEOWNERS` file (you can add agent-platform as joint owners if you want to).

File: test/kitchen/test/integration/dd-agent-runtime-security-test/rspec/dd-agent-runtime-security-test_spec.rb
Comment (albertvaka) [inline test/kitchen/test/integration/dd-agent-runtime-security-test/rspec/dd-agent-runtime-security-test_spec.rb:3]: Nit: The `if` is not needed since we don't run the test on Windows anyway.

**Review by albertvaka — DISMISSED** [review]
LGTM assuming `^TestMkdir$` does what it seems it does and all paths are correct.

Note that until now on kitchen tests we only cared about the OS they ran on, and not the kernel version. If you want to test the EBPF code against several kernels, you might want to use something more specific than `.kitchen_os_centos`, `.kitchen_os_ubuntu`, etc.

But that can be a second iteration, this is much better already than where we were before :D

Edit: +1 to Kylian's comments above ^

File: .gitlab-ci.yml
Comment (lebauce) [reply .gitlab-ci.yml]: Indeed. Fixed

File: test/kitchen/test/integration/dd-agent-runtime-security-test/rspec/dd-agent-runtime-security-test_spec.rb
Comment (lebauce) [reply test/kitchen/test/integration/dd-agent-runtime-security-test/rspec/dd-agent-runtime-security-test_spec.rb]: Removed the `if`

**Review by albertvaka — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Review by safchain — APPROVED** [review]

File: .gitlab-ci.yml
Comment (leeavital) [inline .gitlab-ci.yml:391]: Should `--build-tags=ebpf_bindata` be implicit in the `security-agent.functional-tests` ? Can they be run without ebpf_bindata?

File: pkg/security/ebpf/kprobe_bpf.go
Comment (leeavital) [inline pkg/security/ebpf/kprobe_bpf.go:37]: Why do we retry enabling kprobes? And is this something that should be added to datadog/ebpf? 


Also: EINVAL is a pretty vague error, so this seems like something worth logging at the WARN level.

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb
Comment (leeavital) [inline test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb:64]: What are we using this for ?

File: test/kitchen/kitchen-vagrant-security-agent.yml
Comment (leeavital) [inline test/kitchen/kitchen-vagrant-security-agent.yml:13]: I'm not sure if this is the pattern followed elsewhere, but having a default here seems a little brittle, because if you forget to set TEST_PLATFORMS in gitlab CI nothing will indicate that you made a mistake.

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb
Comment (lebauce) [reply test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb]: We use a /dev/loop* device to create a temporary ext4 filesystem to run the tests

File: pkg/security/ebpf/kprobe_bpf.go
Comment (lebauce) [reply pkg/security/ebpf/kprobe_bpf.go:37]: EINVAL is returned when the sycall does not exist which is not an error for the runtime security agent

File: .gitlab-ci.yml
Comment (lebauce) [reply .gitlab-ci.yml]: Done

File: test/kitchen/kitchen-vagrant-security-agent.yml
Comment (lebauce) [reply test/kitchen/kitchen-vagrant-security-agent.yml:13]: This is the pattern used by the actual kitchen tests. If you forgot to set it in Gitlab CI, it will use these defaults

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb
Comment (guedou) [inline test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb:69]: Is there a reason why this is commented?

File: test/kitchen/test/integration/dd-security-agent-test/rspec/dd-security-agent-test_spec.rb
Comment (guedou) [inline test/kitchen/test/integration/dd-security-agent-test/rspec/dd-security-agent-test_spec.rb:7]: Do you want to also add `expect(output).not_to include("FAIL")`  like in the following `describe` ?

**Review by guedou — APPROVED** [review]
Nice PR. I made some minor comments.

File: .gitlab-ci.yml
Comment (albertvaka) [inline .gitlab-ci.yml:148]: Maybe `if_agent_security_label` would be more descriptive.

Comment (lebauce) [reply .gitlab-ci.yml:148]: It is not used any more (I wanted to trigger the kitchen tests based on labels on the PR but they are not available). Removed

File: test/kitchen/test/integration/dd-security-agent-test/rspec/dd-security-agent-test_spec.rb
Comment (lebauce) [reply test/kitchen/test/integration/dd-security-agent-test/rspec/dd-security-agent-test_spec.rb:11]: I don't think this is useful. If one test fails, it will return something different from 0

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb
Comment (lebauce) [reply test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb:69]: Yes, to not forget to do it when we'll be able to install packages on SUSE

File: pkg/security/ebpf/kprobe_bpf.go
Comment (lebauce) [reply pkg/security/ebpf/kprobe_bpf.go:26]: We can discuss it with the team but I don't think this should be part of the PR

**Pull request merged**
